### PR TITLE
Properly scope fully-qualified class name nodes

### LINKED ISSUE #53: Scope files in the global namespace
I had a simple file:

```php
<?php

require_once __DIR__.'/vendor/autoload.php';

echo (new \Foo\Foo())().PHP_EOL;
```

Where `Foo` was an invokable. Surprisingly the scoping did not prefix `\Foo\Foo()` to `PhpScoperPrefix\Foo\Foo()`.

### PR DESCRIPTION
**Pull request by padraic**
Also fix unrelated bug stripping `\` from start of FQN after concatenating prefixes
closes #53

### CODE REVIEW

File: src/NodeVisitor/FullyQualifiedNamespaceUseScoperNodeVisitor.php [hunk 1]
```diff
@@ -40,7 +40,7 @@ public function enterNode(Node $node)
             && false === ($node->hasAttribute('phpscoper_ignore')
             && true === $node->getAttribute('phpscoper_ignore'))
         ) {
-            return new Name(Name::concat($this->prefix, (string) $node));
+            return new Name('\\'.Name::concat($this->prefix, (string) $node));
```
Comment (padraic) [inline src/NodeVisitor/FullyQualifiedNamespaceUseScoperNodeVisitor.php]: Casting Node to string omits `\` at start so the new Node isn't registered as fully qualified.

File: tests/Scoper/PhpScoperTest.php [hunk 2]
```diff
@@ -594,6 +594,28 @@ public function provideValidFiles()
...
         //
+        // FQN usage for a name
+        //
+        // ====================
+
+        yield '[FQN usage for a name] fully qualified class' => [
+            <<<'PHP'
+<?php
+
+new \Foo\Bar();
```
Comment (theofidry) [inline tests/Scoper/PhpScoperTest.php]: How are we able to tell the difference between `\Closure` and `\Foo`?

**Comment by theofidry** [comment]
Nvm: forgot we are not messing with the global namespace yet so if you have `\Foo` then it's not prefixed

**Comment by padraic** [comment]
Not yet, and the single part visitor I wrote is restricted to handling aliases. What I should do...is open an issue to track.

**Pull request merged**